### PR TITLE
apk: Fix failure when both install and upgrade in same command

### PR DESCRIPTION
**Pull request by tdtrask**
If name list contains an installed package that needs upgrade plus a new package, apk command would fail due to missing space character.

##### SUMMARY
The packages list in install_packages did not insert a space between packages for install and those for upgrade, resulting in failure if both existed in the same command. Since there is no need to track the packages separately, removed the to_upgrade variable and combined into to_install.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/packaging/os/apk.py

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file = 
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION
On a system where wget is not installed and git is installed but not latest:

  tasks:
    - name: Add packages and update
      apk:
        name: git,wget
        state: latest

```
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "failed to install wgetgit", "packages": [], "stderr": "", "stdout": "ERROR: unsatisfiable constraints:\n  wgetgit (missing):\n    required by: world[wgetgit]\n", "stdout_lines": ["ERROR: unsatisfiable constraints:", "  wgetgit (missing):", "    required by: world[wgetgit]"]}

changed: [localhost] => {"changed": true, "msg": "installed git wget package(s)", "packages": ["git", "wget"], "stderr": "", "stdout": "(1/2) Upgrading git (2.11.1-r0 -> 2.11.2-r0)\n(2/2) Installing wget (1.18-r2)\nOK: 815 MiB in 317 packages\n", "stdout_lines": ["(1/2) Upgrading git (2.11.1-r0 -> 2.11.2-r0)", "(2/2) Installing wget (1.18-r2)", "OK: 815 MiB in 317 packages"]}
```

### CODE REVIEW

**Comment by bcoca** [comment]
cc @kbrebanov @tdtrask

File: lib/ansible/modules/packaging/os/apk.py [hunk 1]
```diff
@@ -224,25 +224,24 @@ def upgrade_packages(module, available):
...
             elif state == 'latest' and not query_latest(module, name):
-                to_upgrade.append(name)
-    if to_upgrade:
-        upgrade = True
-    if not to_install and not upgrade:
+                upgrade = True
+                to_install.append(name)
+    if not to_install:
         module.exit_json(changed=False, msg="package(s) already installed")
-    packages = " ".join(to_install) + " ".join(to_upgrade)
```
Comment (samdoran) [inline lib/ansible/modules/packaging/os/apk.py]: A simpler fix would be to do `join(to_install + to_upgrade)`

**Comment by tdtrask** [comment]
@samdoran Yes, that is much simpler. Thanks.
@kbrebanov Preferences? My patch combines to_install and to_upgrade. Or the simpler patch just combines them in the join statement. Both work.

**Comment by kbrebanov** [comment]
@tdtrask I prefer the simpler statement.

**Pull request merged**